algorithms/RIT1-PCA-htcai.ipynb

```diff
@@ -0,0 +1,891 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "0    7285\n",
+       "1      21\n",
+       "Name: 6016, dtype: int64"
+      ]
+     },
+     "execution_count": 8,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# Here is the count of tumors with RIT1\n",
+    "y.value_counts()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Set aside 30% of the data for testing"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "'Size: 20,530 features, 5,114 training samples, 2,192 testing samples'"
+      ]
+     },
+     "execution_count": 9,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# Typically, this can only be done where the number of mutations is large enough\n",
+    "X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, stratify=y, random_state=0)\n",
+    "'Size: {:,} features, {:,} training samples, {:,} testing samples'.format(len(X.columns), len(X_train), len(X_test))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "0    2186\n",
+       "1       6\n",
+       "Name: 6016, dtype: int64"
+      ]
+     },
+     "execution_count": 10,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# Here is the count of tumors with RIT1 in the testing data\n",
+    "y_test.value_counts()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Preprocessing and noise reduction"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "scale_pre = StandardScaler()\n",
+    "X_train_scale = scale_pre.fit_transform(X_train)\n",
+    "X_test_scale = scale_pre.transform(X_test)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "metadata": {
+    "collapsed": false,
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "n_components = 35\n",
+    "pca = PCA(n_components=n_components, random_state=0)\n",
+    "X_train_pca = pca.fit_transform(X_train_scale)\n",
+    "X_test_pca = pca.transform(X_test_scale)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "0.5996\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Percentage of preserved variance\n",
+    "print('{:.4}'.format(sum(pca.explained_variance_ratio_)))"
```

Comment:
So 35 components captures ~60% of variance. Can we increase the number of components until we capture at least 90% of variance?